test/unit/modules/transport.js

```diff
@@ -13,119 +13,356 @@
+					}, function (err) {
+						expect(library.schema.validate.called).to.be.true;
+						expect(err).to.equal('Invalid signatures body');
+						done();
+					});
+				});
+			});
 
-					it('should call __private.receiveSignature');
+			describe('when library.schema.validate succeeds', function () {
 
-					it('should call __private.receiveSignature with signature');
+				describe('for every signature in signatures', function () {
 
-					describe('when __private.receiveSignature fails', function () {
+					it('should call __private.receiveSignature with signature', function (done) {
+						__private.receiveSignature = sinon.stub().callsArg(1);
+						library.schema.validate = sinon.stub().callsArg(2);
+
+						__private.receiveSignatures({
+							signatures: ['SIGNATURE123', 'SIGNATURE456']
+						}, function (err) {
+							expect(library.schema.validate.called).to.be.true;
+							expect(__private.receiveSignature.calledTwice).to.be.true;
+							expect(__private.receiveSignature.calledWith('SIGNATURE123')).to.be.true;
+							expect(__private.receiveSignature.calledWith('SIGNATURE456')).to.be.true;
+							done();
+						});
+					});
 
-						it('should call library.logger.debug with err');
+					describe('when __private.receiveSignature fails', function () {
 
-						it('should call library.logger.debug with signature');
+						it('should call library.logger.debug with err and signature', function (done) {
+							var err = 'Error processing signature: Error message';
+							__private.receiveSignature = sinon.stub().callsArgWith(1, err);
+							library.schema.validate = sinon.stub().callsArg(2);
+							library.logger.debug = sinon.spy();
+
+							__private.receiveSignatures({
+								signatures: ['SIGNATURE123', 'SIGNATURE456']
+							}, function (err) {
+								expect(library.schema.validate.called).to.be.true;
+								// If any of the __private.receiveSignature calls fail, the whole
+								// receiveSignatures operation should fail immediately.
+								expect(__private.receiveSignature.calledOnce).to.be.true;
+								expect(library.logger.debug.calledWith(err, 'SIGNATURE123')).to.be.true;
+								done();
+							});
+						});
 
-						it('should call callback with error');
+						it('should call callback with error', function (done) {
+							var err = 'Error processing signature: Error message';
+							__private.receiveSignature = sinon.stub().callsArgWith(1, err);
+							library.schema.validate = sinon.stub().callsArg(2);
+							library.logger.debug = sinon.spy();
+
+							__private.receiveSignatures({
+								signatures: ['SIGNATURE123', 'SIGNATURE456']
+							}, function (err) {
+								expect(err).to.be.equal(err);
+								done();
+							});
+						});
 					});
 
 					describe('when __private.receiveSignature succeeds', function () {
 
-						it('should call callback with error = undefined');
+						it('should call callback with error = undefined', function (done) {
```

Comment:
So with error = null in this case

Reply:
I it's not strict equality so could be either null or undefined. I'll rename the test case description to reflect that :)